lib/adapters/AbstractAdapter.js

```diff
@@ -185,13 +185,20 @@ class AbstractAdapter extends AbstractReaderWriter {
 		return new Resource(parameters);
 	}
 
-	async _migrateResource(resource) {
+	_migrateResource(resource) {
```

Comment:
changing the function to sync and async return type we have to adjust the usage also, isn't?

Reply:
True. I thought await'ing on a non-promise would not create any overhead. But as I just learned, await will automatically create a resolved Promise in such cases, defeating the whole purpose of this commit 😄 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#description

I added an if-clause to only await if a promise is returned.